test/acceptance/replication/crud_test.go

```diff
@@ -332,6 +332,9 @@ func eventualReplicaCRUD(t *testing.T) {
 		helper.UpdateClass(t, pc)
 	})
 
+	// Ensure consistency on follower nodes by sleeping BEFORE shutting down the leader node
+	time.Sleep(3 * time.Second)
```

Comment:
I would to double check RAFT behaviour here. But in our tests this was the only way to solve this. Basically no data would be present on node_2 without a small sleep here. I'll re-check the logs and understand what is going on.

Reply:
this one i think it's needed because the test actually just configured the number of replica from 1 to 2 so if we stopped 1 immediatly it won't have a chance to broadcast to 2 and it will end with error 
`broadcast: cannot reach enough replicas`

Reply:
But it should be applied in the log and the follower should apply it. Not sure how the process for that works so i need to look into it

Reply:
that is my thoughts: the update will happen on Node one with configuration 
```go
pc.ReplicationConfig = &models.ReplicationConfig{
			Factor: 2,
		}
```

`node 1` given that the replication factor was 1 before, so the node won't update anything but in the middle of replication to `node 2` we shutdown the node and then ask `node 2(follower)` about the data while `node 1` is still down, i think the data will not be there yet until at least node1 is up again with note that we don't restart here until we ask node 2 we just shutdown node 1 and ask node 2 

in other words what we currently do 
update class replication factor  ->  shutdown node1 immediately after update success  ->  ask node 2 about the data -> start node 1 


because node 1 is down while we are asking node 2 about data not replicated yet

Reply:
I agree here the consistency is not about the schema but about the actual data underneath as we discussed in a call earlier. It makes sense to keep the sleep IMO.

Reply:
The question arises: why was it functioning previously without a time delay? There seems to be something missing here. If you examine the sub-test "Configure classes to replicate to node 2," you'll notice the necessity to replicate the object on both nodes after a successful class update. The sleep afterward is not necessary because the main work occurs beforehand

Reply:
It was functionning before with a time delay on the UpdateClass just above https://github.com/weaviate/weaviate/pull/4095/files#diff-8f7596ee44f2376227bba6a51967324fde97a130afe64b71e66191a0cb046747L62.

Small but sufficient to make it work.

Reply:
i think it was succeeding before because we had sleeps within `GetClass` and `UpdateClass`, so it's success was hidden under that sleeps of the Class CRUD

Reply:
The retry should handle the delay, but it seems something may be missing. We can investigate further and create a task to address if necessary.

Reply:
Created a follow-up task: https://semi-technology.atlassian.net/browse/WEAVIATE-701